src/pages/sign-in/index.tsx

```diff
@@ -9,15 +9,31 @@ import { PeerPrepRectLogo } from "~/assets/logo";
 import LoginWithCredentials from "~/components/Login";
 import Link from "next/link";
 import toast from "react-hot-toast";
+import { useState } from "react";
+import { LoadingPage } from "~/components/Loading";
 
 // Signin page with credentials & oauth provider sign in
 type SignInProps = InferGetServerSidePropsType<typeof getServerSideProps>;
 const SignIn = ({ providers }: SignInProps) => {
   const { data: session } = useSession();
+  const [isLoading, setIsLoading] = useState(false);
 
   const onError = (e: Error) => {
-    toast.error(`Failed to sign in: ${e.message}`);
```

Comment:
I think the complaint was that the message was too vague, so in this case we can say something like

```
Wrong email or password
```

Reply:
Best if we don't generalize the error msg to that cos right now we don't know if its cos of Wrong email or password or server down or other reasons, so a vague one is good as long as it doesnt mention CredentialsProvider, i opted to put the message of 
"Failed to sign in \n:
Unable to login with credentials"